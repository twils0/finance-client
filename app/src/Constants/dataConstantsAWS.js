// eslint-disable-next-line
export const actionTypes = {
  SET_AWS_STATUS: 'DATA_AUTH_AWS_STATUS',
  SET_AWS_USER: 'DATA_AUTH_SET_AWS_USER',
  RESET_AWS_STATE: 'DATA_AUTH_RESET_AWS_STATE',
};
