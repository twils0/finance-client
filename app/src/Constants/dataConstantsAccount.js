export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
