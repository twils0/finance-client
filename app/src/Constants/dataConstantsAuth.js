export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
