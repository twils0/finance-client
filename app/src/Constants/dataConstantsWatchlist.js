export const actionTypes = {
  SET_WATCHLIST_STATUS: 'DATA_WATCHLIST_SECURITIES_STATUS',
  SET_SECURITIES_CURRENT: 'DATA_WATCHLIST_SET_SECURITIES_CURRENT',
  SET_SECURITIES_LIST: 'DATA_WATCHLIST_SET_SECURITIES_LIST',
  SET_SECURITIES_ALL: 'DATA_WATCHLIST_SET_SECURITIES_ALL',
  SET_SECURITY: 'DATA_WATCHLIST_SET_SECURITY',
  SET_SECURITY_DATA: 'DATA_WATCHLIST_SET_SECURITY_DATA',
  SET_SECURITY_SUBSCRIBE: 'DATA_WATCHLIST_SET_SECURITY_SUBSCRIBE',
  RESET_WATCHLIST_STATE: 'DATA_ACCOUNT_RESET_WATCHLIST_STATE',
};

export const codeTypeNames = {
  CONFIRM_PHONE: 'confirmPhone',
  CONFIRM_EMAIL: 'confirmEmail',
};

export const statusNames = {
  SECURITIES: 'watchlistStatusSecurities',
  GET_SECURITY_DATA: 'watchlistStatusGetSecurityData',
  UPDATE_SECURITIES: 'watchlistStatusUpdateSecurities',
};
