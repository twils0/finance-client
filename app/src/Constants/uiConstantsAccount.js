export const actionTypes = {
  SET_CURRENT_FORM: 'ACCOUNT_SET_CURRENT_FORM',
  SET_FORM_HEIGHT: 'ACCOUNT_SET_FORM_HEIGHT',
  SET_FORM_EDIT: 'ACCOUNT_SET_FORM_EDIT',
  SET_INPUT_VALUE_ERROR: 'ACCOUNT_SET_INPUT_VALUE_ERROR',
  SET_BUTTON_TEXT: 'ACCOUNT_SET_BUTTON_TEXT',
  SET_BUTTON_VISIBLE: 'ACCOUNT_SET_BUTTON_VISIBLE',
};

export const formNames = {
  CODE: 'codeForm',
  PROFILE: 'profileForm',
  BILLING: 'billingForm',
  CHANGE_PASSWORD: 'changePasswordForm',
  DELETE_ACCOUNT: 'deleteAccountForm',
};

export const buttonNames = {
  PROFILE: 'profileButton',
  BILLING: 'billingButton',
  CHANGE_PASSWORD: 'changePasswordButton',
  DELETE_ACCOUNT: 'deleteAccountButton',
  FIRST: 'firstButton',
  SECOND: 'secondButton',
};

export const buttonTexts = {
  RESEND: 'Resend',
  SUBMIT: 'Submit',
  EDIT: 'Edit',
  UPDATE: 'Update',
  CANCEL: 'Cancel',
  DELETE: 'Delete',
  CONFIRM: 'Confirm',
};

export const inputNames = {
  [formNames.CODE]: {
    CODE_PHONE: 'codeCodePhone',
  },
  [formNames.PROFILE]: {
    NAME: 'profileName',
    EMAIL: 'profileEmail',
    EMAIL_ADDITIONAL: 'profileEmailAdditional',
    PHONE: 'profilePhone',
  },
  [formNames.BILLING]: {
    NAME_ON_CARD: 'billingNameOnCard',
    PROMO_CODE: 'billingPromoCode',
    STRIPE: 'billingStripe',
  },
  [formNames.CHANGE_PASSWORD]: {
    OLD_PASSWORD: 'changePasswordOldPassword',
    NEW_PASSWORD: 'changePasswordNewPassword',
    NEW_PASSWORD2: 'changePasswordNewPassword2',
  },
  [formNames.DELETE_ACCOUNT]: {
    CONFIRM: 'deleteAccountConfirm',
  },
};
