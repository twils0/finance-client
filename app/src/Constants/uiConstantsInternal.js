export const actionTypes = {
  SET_CURRENT_PAGE_BODY: 'INTERNAL_SET_CURRENT_PAGE_BODY',
};

export const pageBodyNames = {
  WATCHLIST: 'watchlistPageBody',
  ACCOUNT: 'accountPageBody',
};

export const buttonNames = {
  WATCHLIST: 'watchlistButton',
  ACCOUNT: 'accountButton',
  LOGOUT: 'logoutButton',
};
