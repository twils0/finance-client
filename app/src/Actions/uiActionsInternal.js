import { actionTypes } from '../Constants/uiConstantsInternal';

export const setCurrentPageBody = (payload) => {
  if (!Object.prototype.hasOwnProperty.call(payload, 'current')) {
    throw new Error(`Please enter a valid 'current' key - ${JSON.stringify(payload)}`);
  }

  return {
    type: actionTypes.SET_CURRENT_PAGE_BODY,
    payload,
  };
};

export default setCurrentPageBody;
