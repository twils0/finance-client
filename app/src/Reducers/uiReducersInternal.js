import { combineReducers } from 'redux';

import { actionTypes, pageBodyNames } from '../Constants/uiConstantsInternal';
import watchlist from './uiReducersWatchlist';
import account from './uiReducersAccount';

export const initialState = {
  utilities: {
    menu: {
      securities: 5, // number of securities in menu
      widthBasis: 600,
      widthBuffer: 60,
      height: 73,
      heightBuffer: 30,
    },
    menuImg: {
      width: 250,
    },
    menuSearchBar: {
      inputLagTime: 200, // time in milliseconds to wait before search
      width: 300,
      height: 45,
      heightBuffer: 10,
    },
    menuSecurity: {
      widthBuffer: 20,
      heightBuffer: 20,
    },
  },
  pageBodies: {
    current: pageBodyNames.WATCHLIST,
    list: [...Object.values(pageBodyNames)],
    [pageBodyNames.WATCHLIST]: { id: pageBodyNames.WATCHLIST },
    [pageBodyNames.ACCOUNT]: { id: pageBodyNames.ACCOUNT },
  },
};

initialState.utilities.menu.mediaWidth = 150
  + 110 // for the searchbar
  + initialState.utilities.menu.widthBasis
  + initialState.utilities.menu.widthBuffer
  + initialState.utilities.menuSearchBar.width
  + initialState.utilities.menu.securities * initialState.utilities.menuSecurity.widthBuffer;

export const utilities = (state = initialState.utilities) => state;

export const pageBodies = (state = initialState.pageBodies, action) => {
  switch (action.type) {
    case actionTypes.SET_CURRENT_PAGE_BODY:
      return {
        ...state,
        ...action.payload,
      };
    default:
      return state;
  }
};

const uiReducersInternal = combineReducers({
  utilities,
  pageBodies,
  watchlist,
  account,
});

export default uiReducersInternal;
