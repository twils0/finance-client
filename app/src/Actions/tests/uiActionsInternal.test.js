import configureMockStore from 'redux-mock-store';
import { initialState } from '../../Reducers/uiReducersInternal';
import { actionTypes } from '../../Constants/uiConstantsInternal';
import { setCurrentPageBody } from '../uiActionsInternal';

const middleware = [];
const mockStore = configureMockStore(middleware);

describe('uiActionsInternal', () => {
  describe('setCurrentPageBody', () => {
    it("fails when not provided a payload with a 'current' key", () => {
      const store = mockStore(initialState);

      const payload = {};

      expect(() => store.dispatch(setCurrentPageBody(payload))).toThrowErrorMatchingSnapshot();
    });

    it('creates the correct action with the correct payload', () => {
      const store = mockStore(initialState);

      const payload = { current: 'testId' };
      const expectedActions = [
        {
          type: actionTypes.SET_CURRENT_PAGE_BODY,
          payload,
        },
      ];

      store.dispatch(setCurrentPageBody(payload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
    });
  });
});
