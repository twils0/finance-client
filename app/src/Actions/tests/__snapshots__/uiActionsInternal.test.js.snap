// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uiActionsInternal setCurrentPageBody fails when not provided a payload with a 'current' key 1`] = `"Please enter a valid 'current' key - {}"`;
